lib/charms/opensearch/v0/opensearch_relation_peer_cluster.py

```diff
@@ -594,19 +603,20 @@ def _s3_credentials(
     def _rel_data(
         self,
         deployment_desc: DeploymentDescription,
-        orchestrators: PeerClusterOrchestrators,
     ) -> Union[PeerClusterRelData, PeerClusterRelErrorData]:
         """Build and return the peer cluster rel data to be shared with requirer sub-clusters."""
-        if rel_err_data := self._rel_err_data(deployment_desc, orchestrators):
-            return rel_err_data
-
         # check that this cluster is fully ready, otherwise put "configuring" in
         # peer rel data for requirers to show a blocked status until it's fully
         # ready (will receive a subsequent
+        cm_nodes = []
         try:
+            cm_nodes = self._fetch_local_cm_nodes(deployment_desc)
+        except OpenSearchHttpError:
+            logger.warning(f"Could not fetch nodes in related {deployment_desc.typ} sub-cluster")
+        finally:
```

Comment:
The `finally` is not needed. We can have a simple return outside of the try/except block